### ISSUE TITLE
rule: `aws_acm_certificate_lifecycle` fails when a dynamic `validation_option` block is used

### ISSUE THREAD

**Issue by davidferguson-cr** [terraform-linters/tflint-ruleset-aws#391]
It seems that the addition of a dynamic `validation_option` block into an `aws_acm_certificate` resource prevents the detection of a perfectly valid `lifecycle` block required by the `aws_acm_certificate_lifecycle` rule.  I understand that `tflint` does not support iterator evaluation on dynamic blocks, but it shouldn't need to in this case.

`main.tf`:
```hcl
locals {
  domain_name       = "foo.bar.com"
  validation_domain = "bar.com"
  validation_option = {
    (local.domain_name) = {
      validation_domain = local.validation_domain
    }
  }
}

resource "aws_acm_certificate" "this" {
  domain_name               = "testing.foo.com"
  validation_method         = "EMAIL"

  dynamic "validation_option" {
    for_each = local.validation_option

    content {
      domain_name       = try(validation_option.value["domain_name"], validation_option.key)
      validation_domain = validation_option.value["validation_domain"]
    }
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

`.tflint.hcl`:
```hcl
plugin "aws" {
  enabled = true
  version = "0.18.0"
  source  = "github.com/terraform-linters/tflint-ruleset-aws"
}
```

`tflint --version`:
```bash
TFLint version 0.42.1
+ ruleset.aws (0.18.0)
+ ruleset.terraform (0.2.0-bundled)
```

`terraform -version`:
```bash
Terraform v1.1.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.36.1
```

**Error:**
```
TFLINT_LOG=debug tflint --config=.tflint.hcl
18:55:56 config.go:126: [INFO] Load config: .tflint.hcl
18:55:56 config.go:253: [DEBUG] Config loaded
18:55:56 config.go:254: [DEBUG]   Module: false
18:55:56 config.go:255: [DEBUG]   Force: false
18:55:56 config.go:256: [DEBUG]   IgnoreModules:
18:55:56 config.go:260: [DEBUG]   Varfiles:
18:55:56 config.go:261: [DEBUG]   Variables:
18:55:56 config.go:262: [DEBUG]   DisabledByDefault: false
18:55:56 config.go:263: [DEBUG]   PluginDir:
18:55:56 config.go:264: [DEBUG]   Format:
18:55:56 config.go:265: [DEBUG]   Rules:
18:55:56 config.go:269: [DEBUG]   Plugins:
18:55:56 config.go:271: [DEBUG]     aws: enabled=true, version=0.18.0, source=github.com/terraform-linters/tflint-ruleset-aws
18:55:56 config.go:298: [INFO] The `terraform` plugin block is not found. Enable the plugin `terraform` automatically
18:55:56 option.go:49: [DEBUG] CLI Options
18:55:56 option.go:50: [DEBUG]   Module: false
18:55:56 option.go:51: [DEBUG]   Force: false
18:55:56 option.go:52: [DEBUG]   IgnoreModules:
18:55:56 option.go:56: [DEBUG]   EnableRules:
18:55:56 option.go:57: [DEBUG]   DisableRules:
18:55:56 option.go:58: [DEBUG]   Only:
18:55:56 option.go:59: [DEBUG]   EnablePlugins:
18:55:56 option.go:60: [DEBUG]   Varfiles:
18:55:56 option.go:61: [DEBUG]   Variables:
18:55:56 option.go:62: [DEBUG]   Format:
18:55:56 loader.go:43: [INFO] Initialize new loader
18:55:56 loader.go:71: [INFO] Module inspection is disabled. Building a root module without children...
18:55:56 loader.go:121: [INFO] Load values files
18:55:56 runner.go:45: [INFO] Initialize new runner for root
18:55:56 discovery.go:90: [DEBUG] Find plugin path: /Users/david.ferguson/.tflint.d/plugins/github.com/terraform-linters/tflint-ruleset-aws/0.18.0/tflint-ruleset-aws
18:55:56 discovery.go:54: [INFO] Plugin `aws` found
18:55:56 [WARN]  go-plugin@v1.4.5/client.go:551: plugin configured with a nil SecureConfig
18:55:56 [DEBUG] go-plugin@v1.4.5/client.go:585: starting plugin: path=/Users/david.ferguson/.tflint.d/plugins/github.com/terraform-linters/tflint-ruleset-aws/0.18.0/tflint-ruleset-aws args=["/Users/david.ferguson/.tflint.d/plugins/github.com/terraform-linters/tflint-ruleset-aws/0.18.0/tflint-ruleset-aws"]
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:593: plugin started: path=/Users/david.ferguson/.tflint.d/plugins/github.com/terraform-linters/tflint-ruleset-aws/0.18.0/tflint-ruleset-aws pid=12059
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:688: waiting for RPC address: path=/Users/david.ferguson/.tflint.d/plugins/github.com/terraform-linters/tflint-ruleset-aws/0.18.0/tflint-ruleset-aws
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:736: using plugin: version=11
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:1030: tflint-ruleset-aws: 18:55:57 [DEBUG] go-plugin@v1.4.5/server.go:401: plugin address: network=unix address=/var/folders/3f/j0hg84bj5pz9fr835lng8ll40000gq/T/plugin1159403055
18:55:57 discovery.go:33: [INFO] Plugin `terraform` is not installed, but the bundled plugin is available.
18:55:57 discovery.go:54: [INFO] Plugin `terraform` found
18:55:57 [WARN]  go-plugin@v1.4.5/client.go:551: plugin configured with a nil SecureConfig
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:585: starting plugin: path=/opt/homebrew/bin/tflint args=["/opt/homebrew/bin/tflint", "--act-as-bundled-plugin"]
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:593: plugin started: path=/opt/homebrew/bin/tflint pid=12061
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:688: waiting for RPC address: path=/opt/homebrew/bin/tflint
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:736: using plugin: version=11
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:1030: tflint: 18:55:57 [DEBUG] go-plugin@v1.4.5/server.go:401: plugin address: network=unix address=/var/folders/3f/j0hg84bj5pz9fr835lng8ll40000gq/T/plugin3448809440
18:55:57 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
18:55:57 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
1 issue(s) found:

Warning: resource `aws_acm_certificate` needs to contain `create_before_destroy = true` in `lifecycle` block (aws_acm_certificate_lifecycle)

  on main.tf line 11:
  11: resource "aws_acm_certificate" "this" {

Reference: https://github.com/terraform-linters/tflint-ruleset-aws/blob/v0.18.0/docs/rules/aws_acm_certificate_lifecycle.md

18:55:57 [DEBUG] go-plugin@v1.4.5/grpc_stdio.go:139: stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
18:55:57 [INFO]  go-plugin@v1.4.5/client.go:646: plugin process exited: path=/opt/homebrew/bin/tflint pid=12061
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:461: plugin exited
18:55:57 [DEBUG] go-plugin@v1.4.5/grpc_stdio.go:139: stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
18:55:57 [INFO]  go-plugin@v1.4.5/client.go:646: plugin process exited: path=/Users/david.ferguson/.tflint.d/plugins/github.com/terraform-linters/tflint-ruleset-aws/0.18.0/tflint-ruleset-aws pid=12059
18:55:57 [DEBUG] go-plugin@v1.4.5/client.go:461: plugin exited
```

Removing the `dynamic "validation_option"` block or converting it to a static block results in no warnings.

**Comment by wata727** [comment]
Fixed in TFLint v0.43.0
https://github.com/terraform-linters/tflint/releases/tag/v0.43.0
